test/unit/Token/SecureParserTest.php

```diff
@@ -0,0 +1,627 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Token;
+
+use DateTimeImmutable;
+use Lcobucci\JWT\Decoder;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\ConstraintViolation;
+use Lcobucci\JWT\Validation\RequiredConstraintsViolated;
+use Lcobucci\JWT\Validation\SignedWith;
+use Lcobucci\JWT\Validation\ValidAt;
+use Lcobucci\JWT\Validation\Validator;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use RuntimeException;
+
+use function uniqid;
+
+/**
+ * @coversDefaultClass \Lcobucci\JWT\Token\SecureParser
+ *
+ * @uses  \Lcobucci\JWT\Token\Parser
```

Comment:
Since `Parser` dependency is hardcoded internally, this test was actually copy-pasted as the two classes, from outside, are not related.